compliance_checker/ioos.py

```diff
@@ -346,6 +348,353 @@ def check_units(self, ds):
+            'publisher_state',
+            'publisher_type',
+            'references',
+        ]
+
+    @check_has(BaseCheck.HIGH)
+    def check_high(self, ds):
+        '''
+        Performs a check on each highly recommended attributes' existence in the dataset
+
+        :param netCDF4.Dataset ds: An open netCDF dataset
+        '''
+        return self.required_atts
+
+    @check_has(BaseCheck.MEDIUM)
+    def check_recommended(self, ds):
+        '''
+        Performs a check on each recommended attributes' existence in the dataset
+
+        :param netCDF4.Dataset ds: An open netCDF dataset
+        '''
+        return self.rec_atts
+
+    def check_standard_name(self, ds):
+        """
+        Wrapper for checking standard names using the CF module.
+        Extends the StandardNameTable to include QARTOD variable
+        standard names.
+        """
+
+        return self.cf1_7.check_standard_name(ds)
+
+    def check_units(self, ds):
+        """
+        Wrapper to check units with the CF module.
+        """
+
+        return self.cf1_7.check_units(ds)
+
+    def check_vars_have_attrs(self, ds):
+        """
+        Using the tuples defined in __init__, check that each variable has
+        the attributes using the corresponding priority.
+
+        :param netCDF4.Dataset: open netCDF4 dataset
+        """
+
+        results = []
+        # NOTE should it also find 'geospatial` variables?
+        for geo_var in get_geophysical_variables(ds):
+            for attr_tuple in self.check_var_attrs:
+                results.append(
+                    self._has_var_attr(
+                        ds,
+                        geo_var,
+                        attr_tuple[0], # attribute name
+                        attr_tuple[0], # attribute name used as 'concept_name'
+                        attr_tuple[1]  # priority level
+                    )
+                )
+
+        return results
+
+
+    def check_single_platform(self, ds):
+        """
+        Verify that a dataset only has a single platform attribute.
+
+        Args:
+            ds (netCDF-4 Dataset): open Dataset object
+
+        Returns:
+            None
+        """
+
+        platform_set = set()
+
+        global_platform = getattr(ds, "platform", None)
+        if global_platform:
+            platform_set.add(global_platform)
+
+        for v in ds.get_variables_by_attributes(platform=lambda x: x is not None):
+            platform_set.add(v.getncattr("platform"))
+
+        msg = f"A dataset may only have one platform; {len(platform_set)} found"
+        if len(platform_set) > 1:
+            return Result(BaseCheck.HIGH, False, "platform", [msg])
+        elif len(platform_set) == 0:
+            return Result(BaseCheck.HIGH, False, "platform", ["A dataset must have a global attribute \"platform\""])
+        else:
+            return Result(BaseCheck.HIGH, True, "platform", [msg])
+        
+    def _check_gts_ingest(self, attr, msg):
```

Comment:
This should probably be implemented as a checker option/ `-O`, e.g. `ioos:gts_ingest_enabled` since not everyone is going to want to submit metadata to GTS.

Reply:
I'm not totally opposed to this, but I already have implemented a conditional to check GTS only if the attribute exists:

https://github.com/ioos/compliance-checker/blob/3767ad9fae1c20ba2faa5c00f92d9f9ae03f76ca/compliance_checker/ioos.py#L577-L596

Is that sub-optimal to a checker option?

Reply:
Upon further thought, I think it's best to run some sub-checks if global attribute `gts_ingest` exists and is "true".